sycl/doc/extensions/proposed/sycl_ext_oneapi_flush.asciidoc

```diff
@@ -0,0 +1,95 @@
+= sycl_ext_oneapi_flush
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:dpcpp: pass:[DPC++]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) 2022-2022 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 5 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+== Status
+
+This is a proposed extension specification, intended to gather community
+feedback.  Interfaces defined in this specification may not be implemented yet
+or may be in a preliminary state.  The specification itself may also change in
+incompatible ways before it is finalized.  *Shipping software products should
+not rely on APIs defined in this specification.*
+
+== Overview
+
+This extension adds a mechanism to signal that a SYCL queue should begin
+executing submitted commands whose dependences have been satisfied.
+
+This can be useful if a SYCL runtime or backend runtime batches
+several commands before passing them to the device for execution.
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_FLUSH` to one of the values defined in the table
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
+=== Flush
+
+ This extension adds the following new member function to the SYCL `queue`
+ class:
+
+ ```
+ class queue {
+   void ext_oneapi_flush();
+ };
+ ```
+
+After calls to `ext_oneapi_flush` return, any previously submitted commands
+on the queue are guaranteed to begin executing once their prequisites have
+been satisfied. Note that this does not guarantee that the commands have
+finished executing when `ext_oneapi_flush` has returned.
```

Comment:
This reads slightly differently from the overview in regards to those submitted commands that don't have their dependencies satisfied by the time the call is made. I think it'd be better to have more alignment in phrasing between these two places (or maybe that's just me not being a native speaker).

Reply:
The "Specification" part of the document is the formal definition of the feature, and I think the text that James has is accurate.

The "Overview" is not the formal definition of the feature, but I agree it could use a tweak.  Perhaps the overview could say something less precise like:

> This extension allows an application to ask for commands on a queue to be executed as soon as possible.  This is useful when the implementation batches several commands before submitting them to the device for execution.